### PR TITLE
Add code coverage and sonarcloud analysis

### LINKED ISSUE #13: Measure and show code coverage
Coveralls, perhaps?

With https://github.com/trautonen/coveralls-maven-plugin ?

### PR DESCRIPTION
**Pull request by jeanpic**
Fixes #13 : Modified workflow to include code coverage and sonarcloud analysis.

I had to change the JDK used to build the project because sonarcloud plugin does not allow JDK 8 use anymore, but the project still targets Java 8. I updated maven.compiler.source and target and removed the configuration in the maven compiler plugin was there a good reason for two different configurations ?

### CODE REVIEW

**Comment by guhur** [comment]
Hi @jeanpic thanks a lot for your contribution! Java 8 is unfortunately a requirement for us. Do you think a work-around would exist to use Java 8?

**Comment by jeanpic** [comment]
What could be done is doing the sonar analysis only with JDK 11 but still build and test with the JDK 8 I suppose. Any specific reason why you need JDK 8 ?

**Comment by guhur** [comment]
Several people who are using the releases are still on Java 8. So basically, we need to provide releases + CI for Java 8.
I think your suggestion is great, but it sounds a bit weird to do only the sonar analysis on JDK11. 

Maybe we could do the following?

JDK 11: sonar analysis + build + test
JDK 8: build + test

Apparently, the [Github CI works fine with multiple versions of JDK8](https://github.com/actions/setup-java#testing-against-different-java-versions)

**Comment by domi41** [comment]
First of all, thanks !  :partying_face: 

> was there a good reason for two different configurations

No.  I experimented with both maven and gradle, and with eclipse presets.  What you saw is probably the BOOP (Bits Out Of Place) of my newcomer's journey.  Thanks for noticing it and fixing it !  It's best to have a single source of truth.

> Any specific reason why you need JDK 8

Some of our early adopters expressed the need for Java 8.
We should make sure that tests pass with that java version, but code coverage can use any.

---

Thanks again !  I'm eager to see our coverage and add some more tests to get to the 90' !

**Review by domi41 — APPROVED** [review]
Doing both java 8 and 11 will either require serious duplication or YAML pointer shenanigans (if those even work at all).

I'm okay for doing the grunt work if you wanna stop there.
I think for the CI to pass we'll need to put some sonarcloud keys in the env.

**Comment by jeanpic** [comment]
Setting Java 8 as source and target should be enough, no need to build it with the JDK 8.

Yes to make it work you'll need to create a sonarcloud account, create a new project, update the project id and organization properties in the pom.xml, add the SONAR_TOKEN secret, and update the hyperlinks for the badges. I can't do it, since I don't have the permissions for this repository...

**Comment by domi41** [comment]
Sonarcloud won't let me import this project.
I think the organization MieuxVoter already exists in sonarcloud and I'm not the admin either, so I can't pursue.

@guhur , do you know who has the keys to the toolshed ?

**Comment by guhur** [comment]
I just added you as an admin to the Mieux Voter group

File: pom.xml [hunk 1]
```diff
@@ -13,8 +13,11 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.compiler.source>1.7</maven.compiler.source>
-    <maven.compiler.target>1.7</maven.compiler.target>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <sonar.projectKey>jeanpic_majority-judgment-library-java</sonar.projectKey>
```
Comment (domi41) [inline pom.xml]:
```suggestion
    <sonar.projectKey>MieuxVoter_majority-judgment-library-java</sonar.projectKey>
```

File: pom.xml [hunk 2]
```diff
@@ -13,8 +13,11 @@
...
+    <sonar.organization>jeanpic</sonar.organization>
```
Comment (domi41) [inline pom.xml]:
```suggestion
    <sonar.organization>mieuxvoter</sonar.organization>
```

**Review by domi41 — CHANGES_REQUESTED** [review]
Let's try this for size, shall we?

**Comment by domi41** [comment]
Thanks for the commits !

We're ready to merge, aren't we?

I see a :

```
Error:  Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.9.0.2155:sonar (default-cli) on project majority-judgment: You're not authorized to run analysis. Please contact the project administrator. -> [Help 1]
```

We have a `SONAR_TOKEN` though.

**Review by domi41 — APPROVED** [review]

**Pull request merged**